libs/server/Resp/RespServerSession.cs

```diff
@@ -351,61 +383,46 @@ private bool ProcessBasicCommands<TGarnetApi>(byte* ptr, RespCommand cmd, ref TG
-                RespCommand.PING => NetworkPING(),
+                RespCommand.PING => count == 0 ? NetworkPING() : ProcessArrayCommands(cmd, subcmd, count, ref storageApi),
                 RespCommand.ASKING => NetworkASKING(),
                 RespCommand.MULTI => NetworkMULTI(),
                 RespCommand.EXEC => NetworkEXEC(),
                 RespCommand.UNWATCH => NetworkUNWATCH(),
                 RespCommand.DISCARD => NetworkDISCARD(),
                 RespCommand.QUIT => NetworkQUIT(),
-                RespCommand.RUNTXP => NetworkRUNTXPFast(ptr),
+                RespCommand.RUNTXP => NetworkRUNTXP(ptr, count),
                 RespCommand.READONLY => NetworkREADONLY(),
                 RespCommand.READWRITE => NetworkREADWRITE(),
-                _ => ProcessArrayCommands(ref storageApi),
+
+                _ => ProcessArrayCommands(cmd, subcmd, count, ref storageApi)
+
             };
             return success;
         }
 
-        private bool ProcessArrayCommands<TGarnetApi>(ref TGarnetApi storageApi)
+        private bool ProcessArrayCommands<TGarnetApi>(RespCommand cmd, byte subcmd, int count, ref TGarnetApi storageApi)
            where TGarnetApi : IGarnetApi
         {
-            if (*(recvBufferPtr + readHead) != '*')
-            {
-                if (bytesRead - readHead < 2) return false;
-                if (*(ushort*)(recvBufferPtr + bytesRead - 2) != 2573)
-                    return false;
-                readHead = bytesRead;
-                while (!RespWriteUtils.WriteResponse(CmdStrings.RESP_ERR, ref dcurr, dend))
-                    SendAndReset();
-                return true;
-            }
+            // Continue reading from the current read head.
+            byte* ptr = recvBufferPtr + readHead;
 
-            var tmp = recvBufferPtr + readHead;
-            if (!RespReadUtils.ReadArrayLength(out int count, ref tmp, recvBufferPtr + bytesRead))
-                return false;
-            readHead += (int)(tmp - (recvBufferPtr + readHead));
+            if (!_authenticator.IsAuthenticated) return ProcessOtherCommands(cmd, subcmd, count, ref storageApi);
 
-            if (!_authenticator.IsAuthenticated) return ProcessOtherCommands(count, ref storageApi);
-
-            var ptr = recvBufferPtr + readHead;
-            var cmd = FastParseArrayCommand(count, recvBufferPtr + readHead);
-            var success = cmd switch
+            var success = (cmd, subcmd) switch
             {
                 (RespCommand.MGET, 0) => NetworkMGET(count, ptr, ref storageApi),
                 (RespCommand.MSET, 0) => NetworkMSET(count, ptr, ref storageApi),
                 (RespCommand.MSETNX, 0) => NetworkMSETNX(count, ptr, ref storageApi),
-                (RespCommand.DEL, 0) => NetworkDEL(count, ptr, false, ref storageApi),
-                (RespCommand.UNLINK, 0) => NetworkDEL(count, ptr, true, ref storageApi),
+                (RespCommand.UNLINK, 0) => NetworkDEL(count, ptr, ref storageApi),
```

Comment:
Where did RespCommand.DEL & RespCommand.EXISTS go? Was that on purpose?

Reply:
Those previously had to be checked twice, because they could be parsed in two different places (for single key and array commands). Now that parsing is happening before execution we are only keeping the invocations in `ProcessBasicCommands()`.